libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/location/ConstantVelocityInterpolator.kt

```diff
@@ -0,0 +1,49 @@
+package com.mapbox.navigation.ui.maps.internal.location
+
+import android.animation.TimeInterpolator
+import android.graphics.Path
+import android.view.animation.PathInterpolator
+import com.mapbox.geojson.Point
+import kotlin.math.pow
+
+/**
+ * A time interpolator that animates between key points at constant velocity.
+ */
+class ConstantVelocityInterpolator(
+    startPoint: Point,
+    keyPoints: Array<Point>
+) : TimeInterpolator {
+
+    private var innerInterpolator: TimeInterpolator = TimeInterpolator { it }
+
+    init {
+        val distances = mutableListOf<Double>()
+        var total = 0.0
+        keyPoints.fold(startPoint) { prevPoint, point ->
+            val d = distanceSqrTo(prevPoint, point)
+            distances.add(d)
+            total += d
+            point
+        }
+
+        if (0 < total) {
+            val path = Path()
+            val step = 1.0f / keyPoints.size
+            var pathTime = 0.0
+            keyPoints.forEachIndexed { index, point ->
+                val deltaTime = distances[index] / total
+                pathTime += deltaTime
+                path.lineTo(pathTime.toFloat(), step * (index + 1))
+            }
+            innerInterpolator = PathInterpolator(path)
```

Comment:
It can be val if you add an else branch here.

Reply:
Great suggestion. Thx.